packages/ra-core/src/controller/field/useReferenceManyFieldController.ts

```diff
@@ -88,15 +89,24 @@ const useReferenceManyFieldController = ({
     sort = { field: 'id', order: 'DESC' },
 }: Options): ReferenceManyProps => {
     const referenceId = get(record, source);
+    const notify = useNotify();
     const { data, ids, total, loading, loaded } = useGetManyReference(
-        resource,
         reference,
         target,
         referenceId,
         { page, perPage },
         sort,
         filter,
-        source
+        resource,
+        {
+            onFailure: error =>
```

Comment:
such side effects should be added by the called, and not be in the hook

Reply:
Should we review the `useListController`, `useEditController`, `useShowController` and `useCreateController` accordingly ?

Reply:
oops, no, you're right; a *controller* hook can dispatch, it's *generic* hooks that should not. Disregard my previous comment.